AK/NoAllocationGuard.h

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2022, kleines Filmröllchen <malu.bertsch@gmail.com>.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+// NOTE: The no-allocation guard relies on the LibC malloc implementation which checks the global s_allocation_enabled.
+//       Therefore, the guard doesn't work in the Kernel, but it has no use case there anyways.
+#if !defined(KERNEL)
+#    include <AK/Forward.h>
+#    include <AK/Noncopyable.h>
+#    include <LibC/mallocdefs.h>
+
+namespace AK {
+
+class NoAllocationGuard {
+    AK_MAKE_NONCOPYABLE(NoAllocationGuard);
+    AK_MAKE_NONMOVABLE(NoAllocationGuard);
+
+public:
+    NoAllocationGuard()
+        : m_allocation_enabled_previously(s_allocation_enabled)
+    {
+        s_allocation_enabled = false;
+    }
+
+    ~NoAllocationGuard()
+    {
+        s_allocation_enabled = m_allocation_enabled_previously;
+    }
+
+    static bool is_allocation_enabled() { return s_allocation_enabled; }
```

Comment:
Is there a reason why this method needs to exist? This seems like a mechanism for enforcing a no-alloc policy. So the code path either confirms to that policy or doesn't, it's not clear to me in what situation you would conditionally check this in real code?

Reply:
Sure, let's get rid of it